File: 백준/Silver/11723. 집합/집합.py
from sys import stdin
m = int(stdin.readline())
BIT=0
while m > 0:
    string = list(map(str,stdin.readline().split()))
    if string[0]=="add":
        num = int(string[1])
        BIT |= (1<<num)
    elif string[0]=="remove":
        num = int(string[1])
        BIT &= ~(1<<num)
    elif string[0]=="check":
        num = int(string[1])
        if BIT & (1<<num):
            print(1)
        else:
            print(0)
    elif string[0]=="toggle":
        num = int(string[1])
        BIT ^= (1<<num)
    elif string[0]=="all":
        BIT = (1<<21)-1
    elif string[0]=="empty":
        BIT = 0
    m-=1

File: 백준/Silver/11723. 집합/집합.cs
using System;
using System.Text;
class Program
{
    static void Main() {
        string s = Console.ReadLine();
        int m = int.Parse(s);
        int BIT = 0;
        StringBuilder sb = new StringBuilder();
        
        while(m != 0)
        {
            string[] input = Console.ReadLine().Split(' ');
            if(input[0] == "add")
            {
                BIT |= (1 << int.Parse(input[1]));
            }
            else if(input[0] == "remove")
            {
                BIT &= ~(1 << int.Parse(input[1]));
            }
            else if(input[0] == "check")
            {
                if(BIT == (BIT | (1 << int.Parse(input[1])))) 
                {
                    sb.Append(1);
                    sb.Append('\n');
                }
                else 
                {
                    sb.Append(0);
                    sb.Append('\n');
                }
            }
            else if(input[0] == "toggle")
            {
                BIT ^= (1 << int.Parse(input[1]));
            }
            else if(input[0] == "all")
            {
                BIT = (1 << 21) - 1;
            }
            else if(input[0] == "empty")
            {
                BIT = 0;
            }
            m--;
        }

        Console.Write(sb);
    }
}